views.py

```diff
@@ -668,6 +668,63 @@ def typesetting_typesetter_download_file(request, assignment_id, file_id):
             'You do not have permission to view this file.',
         )
 
+@decorators.has_journal
+@security.user_can_manage_file
+def typesetting_download_file(request, article_id, file_id):
+    """
+    A view that serves up a file for a given article.
+    """
+    file = get_object_or_404(
+        core_models.File,
+        pk=file_id,
+        article_id=article_id,
+    )
+
+    return files.serve_any_file(
+        request,
+        file,
+        path_parts=('articles', article_id)
+    )
+
+
+@decorators.has_journal
+@security.user_can_manage_file
+def typesetting_delete_file(request, file_id):
+    """
+    Allows users with permission to delete files
+    """
+    file = get_object_or_404(
+        core_models.File,
+        pk=file_id,
+    )
+
+    article = get_object_or_404(
+        submission_models.Article,
+        pk=file.article_id,
+    )
+
+
+    # Check how many galleys there are before deleting them.
+    galley_count = file.galley_set.all().count()
```

Comment:
`file.delete()` returns a dict with a count of all the related objects that have been deleted, we don't need this count here